.kitchen.vagrant.yml

```diff
@@ -1,44 +0,0 @@
----
-driver:
```

Comment:
I agree with your aim to merge the kitchen.yml into a more maintainable structure. We could try to use a base configuration and capture the provider in a local kitchen config (KITCHEN_LOCAL_YAML). This would allow us to:
- kitchen.yml with vagant
- kitchen.do.local.yml for DigitalOcean

The kitchen local approach does not work well with `kitchen-dokken`, since nearly everything needs to be overwritten. We could switch to `kitchen-docker`, but this works by installing ssh into the container while `kitchen-dokken` is not.

The initial idea of switching to kitchen-dokken was to do all integration tests in travis. Therefore `kitchen-dokken` was the best solution for all chef projects (except for os-hardening). It works within the travis without external dependencies and is for free. I agree that we need to have another solution for os-hardening since we are testing kernel parameters, but I am not sure we should switch all other projects.

My main goal is to have a self-contained project that can be fully tested locally. I like the digital ocean idea, since it increases test coverage. I do not think we should restrict testing to digital ocean.

@artem-sidorenko Thoughts?